File: examples/python/Camera/camera_output.py
#!/usr/bin/env python3

import argparse
import cv2
import depthai as dai


parser = argparse.ArgumentParser()
parser.add_argument(
    "--fps_limit", type=float, default=None, help="Limit output FPS (float, optional)"
)
args = parser.parse_args()

# Create pipeline
with dai.Pipeline() as pipeline:
    # Define source and output
    cam = pipeline.create(dai.node.Camera).build()
    videoOutput = cam.requestOutput((640, 400))
    videoQueue = videoOutput.createOutputQueue()

    pipeline.build()

    # Optionally update internal settings
    # Note: xlink bridges are only generated after pipeline.build() is called
    if args.fps_limit is not None:
        xlinkBridge = videoOutput.getXLinkBridge()
        assert xlinkBridge is not None
        assert isinstance(xlinkBridge, dai.node.internal.XLinkOutBridge)
        xlinkBridge.xLinkOut.setFpsLimit(args.fps_limit)

    # Connect to device and start pipeline
    pipeline.start()
    while pipeline.isRunning():
        videoIn = videoQueue.get()
        assert isinstance(videoIn, dai.ImgFrame)
        cv2.imshow("video", videoIn.getCvFrame())

        if cv2.waitKey(1) == ord("q"):
            break

File: examples/cpp/Camera/camera_output.cpp
#include <atomic>
#include <csignal>
#include <iostream>
#include <memory>
#include <opencv2/opencv.hpp>

#include "depthai/depthai.hpp"

std::atomic<bool> quitEvent(false);

void signalHandler(int) {
    quitEvent = true;
}

int main() {
    signal(SIGTERM, signalHandler);
    signal(SIGINT, signalHandler);

    // Create device
    std::shared_ptr<dai::Device> device = std::make_shared<dai::Device>();

    // Create pipeline
    dai::Pipeline pipeline(device);

    // Create nodes
    auto cam = pipeline.create<dai::node::Camera>()->build();
    auto videoQueue = cam->requestOutput(std::make_pair(640, 400))->createOutputQueue();

    // Start pipeline
    pipeline.start();

    while(pipeline.isRunning() && !quitEvent) {
        auto videoIn = videoQueue->get<dai::ImgFrame>();
        if(videoIn == nullptr) continue;

        cv::imshow("video", videoIn->getCvFrame());

        if(cv::waitKey(1) == 'q') {
            break;
        }
    }

    pipeline.stop();
    pipeline.wait();

    return 0;
}